File: self/4-positiveSum.js
// function positiveSum(arr) {
//   let result = 0;

//   for (let i = 0; i < arr.length; i++) {
//     if (arr[i] > 0) result += arr[i];
//   }

//   return result;
// }

const positiveSum = (arr) => arr.reduce((acc, curr) => curr > 0 ? acc + curr : acc, 0);

console.log(positiveSum([-1, 2, 3, 4, -5]));

File: self/golang/4_positive_sum.go
/*
  Sum of positive
  https://www.codewars.com/kata/5715eaedb436cf5606000381
*/

package main

import "fmt"

func PositiveSum(numbers []int) int {
	var result int
	for _, number := range numbers {
		if number > 0 {
			result += number
		}
	}
	return result
}

func main() {
	numbers := []int{1, -4, 7, 12}
	result := PositiveSum(numbers)
	fmt.Println(result)
}
